### PR TITLE
Youtube main overlay

### PR DESCRIPTION
**Pull request by liywjl**
## What does this change?
Adds support for youtube overlays

## Have we considered potential risks?
Overlay blocking loading of youtube videos

## Images

Before click:
Screenshot 2020-10-21 at 15 37 14

After:
Screenshot 2020-10-19 at 16 23 27

### CODE REVIEW

File: src/YoutubeAtom/YoutubeOverlay.tsx [hunk 1]
```diff
@@ -1,52 +1,188 @@
...
+                <div className={`${playButtonStyling} overlay-play-button`}>
+                    <SvgPlay />
+                </div>
+                {duration && (
+                    <div className={videoDurationStyles}>
+                        {formatTime(duration)}
+                    </div>
+                )}
+            </div>
             <YoutubeMeta mediaDuration={duration} pillar={pillar} />
```
Comment (nitro-marky) [inline src/YoutubeAtom/YoutubeOverlay.tsx:185]: Might need to remove this if we are formatting and adding the time in the overlay class

File: src/YoutubeAtom/YoutubeAtom.tsx [hunk 2]
```diff
@@ -56,23 +56,29 @@ export const YoutubeAtom = ({
...
     return (
         <div>
             <MaintainAspectRatio height={height} width={width}>
+                <iframe
+                    title={title}
+                    width={width}
+                    height={height}
+                    id={videoMeta.assetId}
+                    // need to add the origin flag back for production when merging
+                    src={`https://www.youtube.com/embed/${videoMeta.assetId}?embed_config=${embedConfig}&enablejsapi=1&origin=https://www.theguardian.com&widgetid=1&modestbranding=1`}
```
Comment (gtrufitt) [inline src/YoutubeAtom/YoutubeAtom.tsx:68]: Needs doing?

File: src/YoutubeAtom/YoutubeOverlay.tsx [hunk 3]
```diff
@@ -1,52 +1,188 @@
...
+    ${textSans.medium({ fontWeight: 'bold' })};
+    padding-left: ${space[3]}px;
+    color: ${palette['news'][500]};
+`;
+
+const formatTime = (videoDurationInSeconds: number) => {
+    const minutes = Math.floor(videoDurationInSeconds / 60);
+    const seconds = videoDurationInSeconds % 60;
+    return `${minutes}:${seconds < 10 ? `0${seconds}` : seconds}`;
+};
```
Comment (gtrufitt) [inline src/YoutubeAtom/YoutubeOverlay.tsx:85]: Is it worth making this a util and running it through it's paces with a test?

File: src/YoutubeAtom/YoutubeMeta.tsx [hunk 4]
```diff
@@ -77,6 +78,31 @@ const Icon = ({ pillar }: { pillar: Pillar }) => (
     </span>
 );
 
+const svgStyle = css`
+    left: 35%;
+    top: 1%;
```
Comment (gtrufitt) [inline src/YoutubeAtom/YoutubeMeta.tsx:83]: Does this need to be % when you've set the height and width of the button in px? I can't see where these styles are used?

**Comment by shtukas** [comment]
Nice!

File: src/lib/formatTime.test.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { formatTime } from './formatTime';
+
+describe('formatTime', () => {
+    it('should format time to correct string', () => {
+        expect(formatTime(4622)).toBe('01:17:02');
+        expect(formatTime(621)).toBe('10:21');
```
Comment (gtrufitt) [inline src/lib/formatTime.test.ts:6]: Sorry, pedantic, can we add a a `0:xx` (less than a minute) one?

**Review by gtrufitt — APPROVED** [review]
Couple of things, but looks good once these are resolved!

The chromatic snapshot doesn't seem to have the overlay image? It is there in storybook though?

image

I also tested in storybook the enter and play keyboard presses and it didn't seem to start playing the video?

File: src/YoutubeAtom/YoutubeOverlay.tsx [hunk 6]
```diff
@@ -1,52 +1,174 @@
...
+            } catch (e) {
+                console.error(`Unable to play video due to: ${e}`);
+            }
+        }
+    }, [player, isPlayerReady]);
+
+    const onKeydownOverlay = useCallback(
+        (e) => {
+            const spaceKey = 32;
+            const enterKey = 13;
```
Comment (gtrufitt) [inline src/YoutubeAtom/YoutubeOverlay.tsx:135]: I also tested in storybook the enter and play keyboard presses and it didn't seem to start playing the video?

Comment (liywjl) [reply src/YoutubeAtom/YoutubeOverlay.tsx:135]: great point, i tried locally and it wouldn't run, it appears that this is due `origin` now appearing in the iFrame `src`. I have now used a bool to toggle it (not ideal but shouldn't be too disruptive)

**Comment by liywjl** [comment]
> Couple of things, but looks good once these are resolved!
> 
> The chromatic snapshot doesn't seem to have the overlay image? It is there in storybook though?
> 
> image
> 
> I also tested in storybook the enter and play keyboard presses and it didn't seem to start playing the video?

That video is not the one what should have the overlay, so the image itself is a little confusing

**Pull request merged**